### PR TITLE
chore(upgrade): v1.3.0 to v1.4.0 

### LINKED ISSUE #1807: Upgrade to Polkadot Release polkadot-v1.4.0
The Parity team has published new version of Polkadot: `polkadot-v1.4.0`

See https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.4.0

## Upgrade Priority

See [Polkadot Release Page](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.4.0)

## Host Functions

See [Polkadot Release Page](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.4.0)

## Database Migrations

See [Polkadot Release Page](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.4.0)

## Runtime Migrations

See [Polkadot Release Page](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.4.0)

### PR DESCRIPTION
**Pull request by enddynayn**
Upgrade Polkadot-sdk to v1.4.0.

- [Polkadot-sdk release notes](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.4.0)
- Update weights

Closes #1807

### CODE REVIEW

File: node/service/src/chain_spec/frequency.rs [hunk 1]
```diff
@@ -121,13 +121,23 @@ pub fn benchmark_mainnet_config() -> ChainSpec {
...
 	let para_id: ParaId = 2091.into();
-	ChainSpec::from_genesis(
-		// Name
+	ChainSpec::builder(
+		frequency_runtime::WASM_BINARY.expect("WASM binary was not build, please build it!"),
+		Extensions { relay_chain: "polkadot".into(), para_id: para_id.into() },
+	).with_name(
 		"Frequency",
-		// ID
+	).with_id(
```
Comment (enddynayn) [inline node/service/src/chain_spec/frequency.rs:129]: with_protocol_id

File: node/service/src/chain_spec/frequency_paseo.rs [hunk 2]
```diff
@@ -90,74 +90,62 @@ pub fn local_paseo_testnet_config() -> ChainSpec {
...
-		// Extensions
+	ChainSpec::builder(
+		frequency_runtime::WASM_BINARY.expect("WASM binary was not build, please build it!"),
 		Extensions {
 			relay_chain: "paseo-local".into(), // You MUST set this to the correct network!
 			para_id: 2000,
 		},
 	)
+	.with_name("Frequency Local Testnet")
+	.with_id("frequency-paseo-local")
```
Comment (enddynayn) [inline node/service/src/chain_spec/frequency_paseo.rs:101]: with_protocol_id

File: node/service/src/chain_spec/frequency_rococo.rs [hunk 3]
```diff
@@ -36,74 +36,62 @@ pub fn local_rococo_testnet_config() -> ChainSpec {
...
-		// Extensions
+	ChainSpec::builder(
+		frequency_runtime::WASM_BINARY.expect("WASM binary was not build, please build it!"),
 		Extensions {
 			relay_chain: "rococo-local".into(), // You MUST set this to the correct network!
 			para_id: 2000,
 		},
 	)
+	.with_name("Frequency Local Testnet")
+	.with_id("frequency-rococo-local")
```
Comment (enddynayn) [inline node/service/src/chain_spec/frequency_rococo.rs:47]: with_protocol_id

File: node/service/src/chain_spec/frequency_dev.rs [hunk 4]
```diff
@@ -20,74 +20,66 @@ pub fn development_config() -> ChainSpec {
...
+		vec![
+			get_account_id_from_seed::<sr25519::Public>("Bob"),
+			get_account_id_from_seed::<sr25519::Public>("Dave"),
+		],
+		// ParaId
+		1000.into(),
+	))
+	.build()
+
+	// Name
```
Comment (enddynayn) [inline node/service/src/chain_spec/frequency_dev.rs:81]: rm

**Review by wilwade — CHANGES_REQUESTED** [review]
Nice update. One fix, one optional fix, and a separate thought.

File: runtime/frequency/src/lib.rs [hunk 5]
```diff
@@ -913,12 +925,13 @@ impl cumulus_pallet_parachain_system::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type OnSystemEvent = ();
 	type SelfParaId = parachain_info::Pallet<Runtime>;
-	type DmpMessageHandler = ();
+	type DmpQueue = frame_support::traits::EnqueueWithOrigin<(), sp_core::ConstU8<0>>;
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply runtime/frequency/src/lib.rs:928]: I thought the same. However, EnqueueWithOrigin is not implemented on ().

File: node/service/Cargo.toml [hunk 6]
```diff
@@ -21,6 +21,7 @@ jsonrpsee = { workspace = true, features = ["server"] }
 log = { workspace = true }
 serde = { workspace = true, features = ["derive"] }
 tokio = { workspace = true, features = ["macros", "time", "parking_lot"] }
+serde_json = { worspace = true }
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply node/service/Cargo.toml]: Thats weird. I would expect this mistake to cause an error.

[the remark this answers is not in the document]
Comment (wilwade) [reply node/service/Cargo.toml]: It threw a warning only for me, but with a note that it would be an error in the future.

**Review by wilwade — APPROVED** [review]
Working for me!

- Review
- Local run

**Review by saraswatpuneet — APPROVED** [review]
* read through the code changes
* Checked correctness of chain spec builders 
🚢

**Review by aramikm — APPROVED** [review]
Looks good to me!

File: Cargo.toml [hunk 7]
```diff
@@ -53,52 +53,52 @@ serde = { version = "1.0", default-features = false }
...
+cumulus-pallet-parachain-system = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
+cumulus-pallet-session-benchmarking = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
+cumulus-primitives-aura = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
+cumulus-primitives-core = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
+cumulus-primitives-parachain-inherent = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0" }
+cumulus-primitives-timestamp = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
+pallet-collator-selection = { git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
+parachain-info = { package = "staging-parachain-info", git = "https://github.com/paritytech/polkadot-sdk", branch = "release-polkadot-v1.4.0", default-features = false }
```
[the remark this answers is not in the document]
Comment (enddynayn) [reply Cargo.toml:101]: They renamed the package.

[the remark this answers is not in the document]
Comment (aramikm) [reply Cargo.toml:101]: having `staging` in the name is a bit confusing, specially since it's about parachains

**Review by mattheworris — COMMENTED** [review]
- Read through changes
- Interesting changes to the genesis build
Nice work!

File: Cargo.toml (hunk 7 above)
[the remark this answers is not in the document]
Comment (enddynayn) [reply Cargo.toml:101]: I agree. It seems a bit odd choice of name.

**Pull request merged**